### PR TITLE
Improve 'nucypher status events' to allow event filtering

### PR DESCRIPTION
**Pull request by derekpierre**
**Type of PR:**
- [ ] Bugfix
- [X] Feature
- [ ] Documentation
- [ ] Other

**Required reviews:** 
- [ ] 1
- [x] 2
- [ ] 3

**What this does:**
> High-level idea of the changes introduced in this PR. 
> List relevant API changes (if any), as well as related PRs and issues.

`nucypher status events` command can now:
* output to csv file using `--csv` or `--csv-file`
* accept 1 or more event filters in the form `--event-filter name=value`, for example you can filter by staker address.

**Issues fixed/closed:**
> - Fixes #...

Related to #2548 , #2561 .

**Why it's needed:**
> Explain how this PR fits in the greater context of the NuCypher Network.
> E.g., if this PR address a `nucypher/productdev` issue, let reviewers know!

This allows for ease of accounting based on events emitted by smart contracts.

For example
```
$ nucypher status events --contract-name StakingEscrow  --provider <provider> --event-filter staker=<staker_address>  --from-block 0 --to-block 11876507 --csv --event-name Minted
```
**Notes for reviewers:**
> What should reviewers focus on? 
> Is there a particular commit/function/section of your PR that requires more attention from reviewers?

* Try `nucypher stake events` and `nucypher status events` commands
* Proofread docs

### CODE REVIEW

**Comment by derekpierre** [comment]
@cygnusv if `nucypher status events` now has this function, is there any reason to keep `nucypher stake events`?...both CLI commands can achieve the same thing; `nucypher status events` does not need the stakeholder.json configuration file to work.

**Comment by KPrasch** [comment]
@derekpierre - Looks like CI is failing due to exceeding the probationary period.

What do you think of extending it another month anticipating DAO proposal #1?

```
nucypher.blockchain.eth.actors.ActorError: The requested duration for this policy (until 2021-03-02 14:58:20.836679+00:00) exceeds the probationary period (2021-02-28 23:59:59+00:00).
```

File: docs/source/references/network_events.rst [hunk 1]
```diff
@@ -0,0 +1,200 @@
+Network Events
```
Comment (KPrasch) [inline docs/source/references/network_events.rst:1]: 👏🏻 Well done - this document is beautiful.

File: docs/source/references/network_events.rst [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+
+For a full list of CLI options, run:
+
+.. code::
+
+    $ nucypher status events --help
+
+
+For example, to view the staking rewards received by all Stakers in the current period, run:
```
Comment (KPrasch) [inline docs/source/references/network_events.rst:28]: Turning these sections into linkable subheaders might make information easier to find for specific use cases.

File: docs/source/references/network_events.rst [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    Retrieving events from block 0 to latest
+
+    --------- PolicyManager Events ---------
+
+    PolicyManager::PolicyCreated events written to PolicyManager_PolicyCreated_2021-02-24_20-02-32.csv
+
+
+.. code::
+
+    $ nucypher status events --provider <PROVIDER URI> --contract-name PolicyManager --event-name PolicyCreated --from-block 0 --csv-file ~/Policy_Events.csv
```
Comment (KPrasch) [inline docs/source/references/network_events.rst:156]: Expanding a bit more, listing several latest events for a given staker when using `nucypher status stakers --staking-address ...` would be helpful for individual node operators who monitor their stake status using the CLI.

File: nucypher/cli/utils.py [hunk 4]
```diff
@@ -218,3 +225,47 @@ def deployer_pre_launch_warnings(emitter: StdoutEmitter, etherscan: bool, hw_wal
         emitter.echo(ETHERSCAN_FLAG_ENABLED_WARNING, color='yellow')
     else:
         emitter.echo(ETHERSCAN_FLAG_DISABLED_WARNING, color='yellow')
+
+
+def parse_event_filters_into_argument_filters(event_filters: Tuple) -> Dict:
```
Comment (KPrasch) [inline nucypher/cli/utils.py:230]: A description of the expected input format as part of a docstring will be helpful here for future maintenance.

File: nucypher/cli/utils.py [hunk 5]
```diff
@@ -218,3 +225,47 @@ def deployer_pre_launch_warnings(emitter: StdoutEmitter, etherscan: bool, hw_wal
...
+    return argument_filters
+
+
+def retrieve_events(emitter: StdoutEmitter,
+                    agent: EthereumContractAgent,
+                    event_name: str,
+                    from_block: BlockIdentifier,
+                    to_block: BlockIdentifier,
+                    argument_filters: Dict,
+                    csv_output_file: Optional[str] = None):
```
Comment (KPrasch) [inline nucypher/cli/utils.py:250]: Include a return type annotation?

**Review by KPrasch — APPROVED** [review]
This is looking great and built with a purpose!

File: docs/source/references/network_events.rst (hunk 3 above)
Comment (derekpierre) [reply docs/source/references/network_events.rst:156]: ✔️

File: nucypher/cli/utils.py (hunk 4 above)
Comment (derekpierre) [reply nucypher/cli/utils.py]: ✔️

File: nucypher/cli/utils.py (hunk 5 above)
Comment (derekpierre) [reply nucypher/cli/utils.py]: ✔️

**Review by vzotova — APPROVED** [review]

**Pull request merged**